nvm.sh

```diff
@@ -3227,11 +3239,13 @@ nvm_auto() {
       nvm install >/dev/null
     fi
   elif [ "_$NVM_MODE" = '_use' ]; then
-   VERSION="$(nvm_resolve_local_alias default 2>/dev/null || nvm_echo)"
-    if [ -n "$VERSION" ]; then
-      nvm use --silent "$VERSION" >/dev/null
-    elif nvm_rc_version >/dev/null 2>&1; then
-      nvm use --silent >/dev/null
+    if [ "_$NVM_CURRENT" = '_system' ]; then
```

Comment:
this seems like a change that could/should go in a separate PR - it seems like a separate issue from the question of prepending vs altering the PATH.

Additionally, if you want to have a noop when it's already using an `nvm`-managed node, that should probably be done before `nvm_auto` is called at all, in `nvm_process_parameters`?